packages/beacon-node/src/chain/chain.ts

```diff
@@ -505,6 +506,36 @@ export class BeaconChain implements IBeaconChain {
     }
   }
 
+  prunePastInvalidSszObjects(): void {
+    const invalidObjectsDir = this.opts.persistInvalidSszObjectsDir ?? "invalid_ssz_objects";
+    const invalidObjectsRetentionDays = this.opts.persistInvalidSszObjectsRetention ?? 7;
+    this.logger.debug("Prunning invalid ssz objects", {invalidObjectsDir, invalidObjectsRetentionDays});
+
+    const retentionDays = invalidObjectsRetentionDays + 1;
+    const DAYS_TO_MS = 24 * 60 * 60 * 1000;
+    const today = new Date();
+    const basePath = invalidObjectsDir;
+
+    const datesPastRetention = (date: string): boolean =>
+      new Date(date).getTime() < today.getTime() - retentionDays * DAYS_TO_MS;
+    fs.readdir(basePath, (err, list) => {
+      if (!err) {
```

Comment:
Might be good for readability to align nested error, success cases consistently here (eg: error first, then success or vice versa) or just use promise-based API.